### ISSUE TITLE
Make it work on FreeBSD

### ISSUE THREAD

**Issue by ahmetalpbalkan** [Azure/custom-script-extension-linux#69]
This handler binary currently does not work on FreeBSD. So we need to:

1. Compile for FreeBSD and bundle it as well.
2. Modify `custom-script-shim` to determine freebsd case and launch the appropriate binary.

**Comment by boumenot** [comment]
Does the FreeBSD image on Azure ship with support for Linux emulation enabled by default?  If so, the extension should just work.

**Comment by lukeb1961** [comment]
is this meant to work?  I created a custom script on Ubuntu and CentOS, but FreeBSD just hangs and never comes back.. log shows nothing.

**Comment by boumenot** [comment]
No, it is not meant to work out-of-the-box.  (PRs are gladly accepted.)  You can try enabling Linux emulation as I mentioned previously, but I have not verified that works.

**Comment by johanburati** [comment]
First there is an issue with `custom-script-shim`, it returns an error when trying to install the extension:
```
2018/03/11 08:18:23.038754 ERROR Event: name=Microsoft.Azure.Extensions.customScript, op=Install, message=[ExtensionError] Non-zero exit code: 127, bin/custom-script-shim install, duration=0
```
The shebang is set to `#!/bin/bash` and `bash` is not installed by default on FreeBSD,
you will have to run `pkg install bash` to install however it installs it in `/usr/local/bin/bash` so you will have to create a soft link to get going, also there is already a [commit](https://github.com/Azure/custom-script-extension-linux/commit/8b64522ae3c3d66ef046b52e0cd6fc7895469501) to fix that so that should be fixed in the next release.

I have configured Linux binary compatibility as per this [link]((https://www.freebsd.org/doc/handbook/linuxemu-lbc-install.html) but I am getting a core dump when the bash script try to execute the handler:
```
./custom-script-shim: line 78:  1394 Segmentation fault      (core dumped) "$bin" $@
```
I have re-compile `custom-script-extension` with `GOOS=freebsd` and it's running, so I think the only way to to deliver a different binary on FreeBSD.

**Comment by KevDBG** [comment]
Hello,

Any update for FreeBSD ? I just try and it doesn't work.

Thanks
